src/backend/distributed/planner/recursive_planning.c

```diff
@@ -546,7 +551,16 @@ BuildSubPlanResultQuery(Query *subquery, uint64 planId, uint32 subPlanId)
 		newTargetEntry = makeNode(TargetEntry);
 		newTargetEntry->expr = (Expr *) functionColumnVar;
 		newTargetEntry->resno = columnNumber;
-		newTargetEntry->resname = columnName;
+		if (cteAliasCount >= columnNumber)
```

Comment:
I cannot follow what does `if (cteAliasCount >= columnNumber)` mean.  

Did we mean `if (cteAliasCount > 0)` such that alias is provided?

Also,  `cteAliasCount` should probably not contain `cte` in the name.

Reply:
renamed the variable.

apparently you can write queries like
```sql
with cte(a, b) as ( select x, y, z from t) select * from cte;
```
cte defines only 2 columns even if inner query has 3. In such cases the output looks like
```sql
 a | b | z
---+---+---
```

Therefore, we only rename the column if an alias is specified, otherwise its own name is used as before.

Reply:
Perhaps I should write that down as a comment

Reply:
> cte defines only 2 columns even if inner query has 3. 

Oh, that's crazy 😄